src/app/dashboard-widgets/work-item-widget/work-item-widget.component.ts

```diff
@@ -48,15 +49,37 @@ export class WorkItemWidgetComponent implements OnInit {
     yData: []
   };
 
-  constructor(private contexts: Contexts,
-              private workItemService: WorkItemService) {
+  constructor(
+    private contexts: Contexts,
+    private workItemService: WorkItemService
+  ) { }
+
+  ngOnInit(): void {
+    this.subscriptions.push(this.contexts.current.subscribe(context => {
+      this.contextPath = context.path;
+      this.resetWorkItemCounts();
+      this.updateWorkItems();
+    }));
+  }
+
+  ngOnDestroy(): void {
+    this.subscriptions.forEach(sub => {
+      sub.unsubscribe();
+    });
   }
 
-  ngOnInit() {
-    this.contextPath = this.contexts.current.map(context => context.path);
+  private resetWorkItemCounts(): void {
+    this.myWorkItemsCount = this.myWorkItemsOpen = this.myWorkItemsInProgress = this.myWorkItemsResolved = 0;
+  }
+
+  private updateWorkItems(): void {
```

Comment:
What do you think of making this not `private` so you can test it?

Reply:
I'll try and write a couple unit tests without changing access modifier

Reply:
Update: I've added unit tests that test the behaviour of the `updateWorkItems` method